compiler/ballerina-parser/src/main/java/io/ballerinalang/compiler/internal/parser/BallerinaParserErrorHandler.java

```diff
@@ -1358,10 +1367,14 @@ private ParserRuleContext getNextRule(ParserRuleContext currentCtx, int nextLook
                     endContext(); // end parameter-list
                 }
                 if (parentCtx == ParserRuleContext.NIL_TYPE_DESCRIPTOR) {
-                endContext();
-                // After parsing nil type descriptor all the other parsing is same as next rule of simple type
-                return getNextRuleForTypeDescriptor();
-            }
+                    endContext();
+                    // After parsing nil type descriptor all the other parsing is same as next rule of simple type
+                    return getNextRuleForTypeDescriptor();
+                }
+                if (parentCtx == ParserRuleContext.NIL_LITERAL) {
+                    endContext();
```

Comment:
Eventhough the `endContext()` is called here, `startContext()` is not called at the first switch-case of `getNextRule()` method.